### PR TITLE
update bundled Lark

### LINKED ISSUE #1064: task advantage of new Lark
Once PR #973 lands, we can:

* remove the workarounds for bugs in old versions of Lark
* refactor to take advantage of new features

This issue is to select a new minimum version, enumerate the changes, evaluate those changes & decided which to do and not do, and then implement accordingly.

### PR DESCRIPTION
**Pull request by lucaudill**
~~closes~~ related to #1064

### CODE REVIEW

**Review by reidpr — COMMENTED** [review]
Discussed offline.

File: packaging/fedora/upstream.spec [hunk 1]
```diff
@@ -43,7 +43,7 @@ Requires:      %{name}
 Requires:      python3
 Requires:      python%{python3_pkgversion}-lark-parser
 Requires:      python%{python3_pkgversion}-requests
-Provides:      bundled(python%{python3_pkgversion}-lark-parser) = 0.11.3
+Provides:      bundled(python%{python3_pkgversion}-lark-parser) = 1.1.8
```
Comment (reidpr) [inline packaging/fedora/upstream.spec:46]: I don't know what this file is about — something related to Jordan’s out-of-tree RPM packaging — and it looks pretty stale, so I think this PR should probably not change it. `charliecloud.spec` is the one we test.

**Review by reidpr — COMMENTED** [review]
RPMs appear to build; sending back to you.

**Review by reidpr — COMMENTED** [review]
The change looks great but it doesn’t solve the linked issue (#1064), which is to update the code to *require* the updated Lark, taking advantage of any new features and removing workarounds for old bugs.

I’m inclined to merge this but remove the link. Opinions?

**Comment by lucaudill** [comment]
Omg you're right. That's what I get for leaving a PR open over the winter closure.

Your suggestion of unlinking the issue and merging sounds good to me, I can address #1064 in a separate PR.

**Pull request merged**